src/lib/inputs/InputNumber.svelte

```diff
@@ -0,0 +1,397 @@
+    export let isLoading: boolean = false;
+    export let min: number = -10000;
+    export let max: number = 10000;
+    export let step: number | "any" = 1;
+
+    // Event dispatcher
+    const dispatch = createEventDispatcher();
+
+    // Variables
+    let containerClassList = ["input"];
+    let containerClassString = "";
+    let readOnly: boolean = false;
+
+    // Lifecycle Hooks
+    onMount(() => {
+        containerClassList.push(size);
+        containerClassList.push(variant);
+        if (isLoading) {
+            containerClassList.push("loading");
+        } else if (isError) {
+            containerClassList.push("error");
+        }
+        containerClassString = containerClassList.join(" ");
+
+        if (value < min) {
+            value = min;
+        } else if (value > max) {
+            value = max;
+        }
+    });
+
+    // Reactive Statements
+    $: {
+        containerClassList = ["input", size, variant];
+        if (isLoading) {
+            containerClassList.push("loading");
+        } else if (isError) {
+            containerClassList.push("error");
+        }
+        containerClassString = containerClassList.join(" ");
+
+        readOnly = disabled || isLoading || isError;
+    }
+
+    $: inputStyle = `
+  background: ${disabled || isError || isLoading ? "#ccc" : background};
+  --color: ${color};
+  color: ${color};
+  ${style}
+`;
+
+    // Handle input event
+    const handleInput = (e: Event) => {
+        if (e.target instanceof HTMLInputElement) {
+            let inputValue = parseInt(e.target.value);
+
+            if (inputValue < min) {
+                inputValue = min;
+            } else if (inputValue > max) {
+                inputValue = max;
+            }
+
+            value = inputValue;
+            e.target.value = value.toString();
+            dispatch("input", e);
+        }
+    };
+
+    // Increment Value
+    const incrementValue = () => {
+        if (isLoading || isError || disabled) return;
+
+        if (Number.isNaN(value)) {
+            value = 0;
+        }
+        let newValue =
+            value + (typeof step === "number" ? step : parseInt(step));
+        if (newValue > max) {
+            newValue = max;
+        }
+        value = newValue;
+    };
+
+    // Decrement Value
+    const decrementValue = () => {
+        if (isLoading || isError || disabled) return;
+
+        if (Number.isNaN(value)) {
+            value = 0;
+        }
+        let newValue =
+            value - (typeof step === "number" ? step : parseInt(step));
+        if (newValue < min) {
+            newValue = min;
+        }
+        value = newValue;
+    };
+
+    // Handle Long Press
+    let incrementTimeout: any;
+    let decrementTimeout: any;
+    let incrementInterval: any;
+    let decrementInterval: any;
+
+    const startIncrementing = () => {
+        incrementTimeout = setTimeout(() => {
+            incrementInterval = setInterval(incrementValue, 100);
```

Comment:
I wonder for all of the set time outs if there is any situation we would want to let the developer set this value as a prop or if that's 'too much'

Reply:
This sounds more like a case by case feature request, tbh.